### PR TITLE
Only billable projects/clients should be displayed on the list.

### PR DESCRIPTION
**Pull request by mayankagnihotri7**
[Notion Card](https://www.notion.so/saeloun/Accounts-Aging-report-Only-billable-projects-clients-should-be-displayed-on-the-list-d01d63e291f94bf280943f957f26b22e)

- Only billable projects/clients should be displayed on the list. Previously we used to show the clients whose projects are not billable and overdue as well.

https://user-images.githubusercontent.com/57438322/219658528-faf6874d-eaeb-4577-bc23-7fc14419663e.mp4

### CODE REVIEW

File: app/services/reports/accounts_aging/fetch_overdue_amount.rb [hunk 1]
```diff
@@ -47,7 +47,9 @@ def total_amount_due_during (invoices, duration)
       end
 
       def clients
-        @_clients ||= current_company.clients.kept.order(name: :asc).uniq
+        @_clients ||= current_company.clients.joins(:projects).where(
+          projects: { billable: true }
```
Comment (shalapatil) [inline app/services/reports/accounts_aging/fetch_overdue_amount.rb:51]: Please add specs for the same.

**Review by shalapatil — CHANGES_REQUESTED** [review]

**Comment by mayankagnihotri7** [comment]
@shalapatil I've added the specs as requested. Please review.

File: spec/requests/internal_api/v1/reports/accounts_aging/index_spec.rb [hunk 2]
```diff
@@ -72,6 +75,26 @@
         expect(json_response["report"]["total_amount_overdue"]["ninety_plus_days"]).to eq("400.0")
         expect(json_response["report"]["total_amount_overdue"]["total"]).to eq("1500.0")
       end
+
+      it "returns only clients having billable projects" do
+        expect(json_response["report"]["clients"][1]["id"]).to eq(client1.id)
+      end
```
Comment (shalapatil) [inline spec/requests/internal_api/v1/reports/accounts_aging/index_spec.rb:81]: it should also have client2 also, right?
check for all the valid clients

Comment (mayankagnihotri7) [reply spec/requests/internal_api/v1/reports/accounts_aging/index_spec.rb:81]: Right! I'll check for it as well.

**Comment by mayankagnihotri7** [comment]
@shalapatil I've made the requested changes. Please review.

File: spec/requests/internal_api/v1/reports/accounts_aging/index_spec.rb (hunk 2 above)
Comment (shalapatil) [reply spec/requests/internal_api/v1/reports/accounts_aging/index_spec.rb:82]: expect(json_response["report"]["clients"].pluck("id")).to eq(client2.id, client1.id)

**Review by shalapatil — COMMENTED** [review]
One small suggestion, rest LGTM

**Comment by mayankagnihotri7** [comment]
@shalapatil I've made the requested changes. Please review.

**Review by shalapatil — APPROVED** [review]

**Pull request merged**